app/assets/javascripts/application.js

```diff
@@ -27,9 +27,13 @@ var searchBoxValue = searchBox.val();
 searchBox.keyup(_.debounce(function () {
   var input = $(this)
   var newValue = input.val();
+  var url_string = window.location.href;
+  var url = new URL(url_string);
+  var fullscr = url.searchParams.get("fullscr");
+  var perPage = url.searchParams.get("per_page");
   if (searchBoxValue !== newValue) {
     input.submit();
-    history.pushState({}, '', '?' + input.serialize());
+    history.pushState({}, '', '?' + input.serialize() + '&fullscr=' + fullscr + '&per_page=' + perPage);
```

Comment:
I think it'd be cleaner to use https://api.jquery.com/serialize/ for this. Maybe on the form.